src/VisualStudio/CSharp/Test/F1Help/F1HelpTests.cs

```diff
@@ -1228,5 +1228,138 @@ void M()
     object Goo() => null;
 }", "discard");
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.F1Help)]
+        public async Task TestChecked_01()
+        {
+            await Test_KeywordAsync(
+@"public class C
+{
+    void goo()
+    {
+        chec[||]ked
+        {
+        }
+    }
+}", "checked");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.F1Help)]
+        public async Task TestChecked_02()
+        {
+            await Test_KeywordAsync(
+@"public class C
+{
+    int goo()
+    {
+        return chec[||]ked(0);
+    }
+}", "checked");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.F1Help)]
+        public async Task TestChecked_03()
+        {
+            await Test_KeywordAsync(
+@"public class C
+{
+    C operator chec[||]ked -(C x) {}
+}", "checked");
```

Comment:
This is not the ideal behavior I think. When the same keyword is valid in different contexts, we add f1_keywords to dotnet/docs for the relevant page and redirect the user to the relevant page. For this case, I'd expect the result to be `checkedOperator` (but before that, it needs to be in dotnet/docs first).

Since this PR adds tests only and doesn't change behavior (and also it's not yet documented in dotnet/docs), no product changes are needed for now. But it's better if they're done later.

Consider tracking that in the test plan.

Reply:
agreed that we should track the new docs page in the test plan.  We should still take this PR though since a) we had no tests for the existing checked operator before and b) this asserts the current behavior of f1 help on the operator.